docs/struct.rst

```diff
@@ -0,0 +1,69 @@
+Working with BigQuery STRUCT data
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The BigQuery `STRUCT data type
+<https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#struct_type>`_
+provided data that are collections of named fields.
+
+`sqlalchemy-bigquery` provided a STRUCT type that can be used to
+define tables with STRUCT columns:
+
+.. literalinclude:: samples/snippets/STRUCT.py
+   :language: python
+   :dedent: 4
+   :start-after: [START bigquery_sqlalchemy_create_table_with_struct]
+   :end-before: [END bigquery_sqlalchemy_create_table_with_struct]
+
+`STRUCT` types can be nested, as in this example.  Struct fields can
+be defined in two ways:
+
+- Fields can be provided as keyword arguments, as in the `cylinder`
+  and `horsepower` fields in this example.
+
+- Fields can be provided as name-type tuples provided as positional
+  arguments, as with the `count` and `compression` fields in this example.
+
+STRUCT columns are automatically created when existing database tables
+containing STRUCT columns are introspected.
+
+Struct data are represented in Python as Python dictionaries:
+
+.. literalinclude:: samples/snippets/STRUCT.py
+   :language: python
+   :dedent: 4
+   :start-after: [START bigquery_sqlalchemy_insert_struct]
+   :end-before: [END bigquery_sqlalchemy_insert_struct]
+
+When querying struct fields, you can use attribute access syntax:
+
+.. literalinclude:: samples/snippets/STRUCT.py
+   :language: python
+   :dedent: 4
+   :start-after: [START bigquery_sqlalchemy_query_struct]
+   :end-before: [END bigquery_sqlalchemy_query_struct]
+
+or mapping access:
+
+.. literalinclude:: samples/snippets/STRUCT.py
+   :language: python
+   :dedent: 4
+   :start-after: [START bigquery_sqlalchemy_query_getitem]
+   :end-before: [END bigquery_sqlalchemy_query_getitem]
+
+and field names are case insensitive:
```

Comment:
Is this because of a client-side convenience? Or is it BigQuery API behavior? Is it possible to have two struct fields that only differ by case?

Reply:
> Is this because of a client-side convenience?

It is convenient. :)  But that's not the reason.

> Or is it BigQuery API behavior?

Yes and also SQL.

> Is it possible to have two struct fields that only differ by case?

no

```
>>> client.query("create table sample_struct (s struct<x INT64, X int64>)").result()
Traceback (most recent call last):
...
google.api_core.exceptions.BadRequest: 400 Field s.X already exists in schema

(job ID: 040b2a36-9157-47f4-a1f4-c7000dfb3e9c)

              -----Query Job SQL Follows-----               

    |    .    |    .    |    .    |    .    |    .    |
   1:create table sample_struct (s struct<x INT64, X int64>)
    |    .    |    .    |    .    |    .    |    .    |
```